lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -300,6 +300,23 @@ static StringRef getVerilogDeclWord(Operation *op) {
   return isProcedural ? "automatic logic" : "wire";
 }
 
+/// Return the name of a value without using the name map.  This is needed when
+/// looking into an instance from a different module as happens with bind.  It
+/// may return "" when unable to determine a name.
+static StringRef getNameRemotely(Value &value,
+                                 ArrayRef<ModulePortInfo> modulePorts) {
```

Comment:
This approach isn't goign to work.  In addition to missing some things, it is missing the `legalizeName` call through.

I think that bind emission will need a two phase approach: when printing the "injected into context" of a bind, we need to capture the names etc as strings, then emit them later at the right place.  This will also let you use emitExpression to get more complicated things and intersect with the temporary naming stuff correctly.

Reply:
The things it misses are emitted as wires with known names later at the instance.  This just gathers things with known names. 
 The problem with two passes is the bind may appear before module containing the instance and we would have to go arbitrarily back to patch it up.  since binds statements can appear inside modules, it's not just enough to process top-level binds after modules.